runtime-modules/project-token/src/lib.rs

```diff
@@ -800,6 +801,239 @@ decl_module! {
+        /// - token total supply and amount value must be s.t. `eval` function doesn't overflow
+        ///
+        /// Postconditions
+        /// - token `bonding_curve` activated with specified parameters
+        /// - `amount` CRT minted into account (which is created if necessary and existential deposit added to it)
+        /// - respective JOY amount transferred from user balance to amm treasury account
+        /// - percentage of the minted CRT transferred to the token content creator (creator reward)
+        /// - event deposited
+        #[weight = 100_000_000] // TODO: adjust weight
+        fn bond(origin, token_id: T::TokenId, member_id: T::MemberId, amount: <T as Config>::Balance, deadline: Option<<T as timestamp::Config>::Moment>, slippage_tolerance: Option<(Permill, JoyBalanceOf<T>)>) -> DispatchResult {
+            if amount.is_zero() {
+                return Ok(()); // noop
+            }
+
+            let sender = ensure_signed(origin.clone())?;
+
+            T::MemberOriginValidator::ensure_member_controller_account_origin(
+                origin,
+                member_id
+            )?;
+
+            let token_data = Self::ensure_token_exists(token_id)?;
+            let curve = token_data.bonding_curve.ok_or(Error::<T>::NotInAmmState)?;
+
+            let user_account_data_exists = AccountInfoByTokenAndMember::<T>::contains_key(token_id, &member_id);
+            let amm_reserve_account = Self::module_bonding_curve_reserve_account(token_id);
+            let price = curve.eval::<T>(amount, token_data.total_supply, BondOperation::Bond)?;
+            let bloat_bond = Self::bloat_bond();
+
+            let joys_required = if !user_account_data_exists {
+                price.saturating_add(bloat_bond)
+            } else {
+                price
+            };
+
+            Self::ensure_can_transfer_joy(&sender, joys_required)?;
+
+            // slippage tolerance check
+            if let Some((slippage_tolerance, desired_price)) = slippage_tolerance {
+                ensure!(price.saturating_sub(desired_price) < slippage_tolerance.mul_floor(desired_price), Error::<T>::SlippageToleranceExceeded);
+            }
+
+            // timestamp deadline check
+            if let Some(deadline) = deadline {
+                ensure!(<timestamp::Pallet<T>>::get() < deadline, Error::<T>::DeadlineExpired);
+            }
+
+            // == MUTATION SAFE ==
+
+            let creator_member_id = token_data.creator_member_id;
+            let creator_amount = curve.creator_reward.mul_floor(amount);
+            if AccountInfoByTokenAndMember::<T>::contains_key(token_id, creator_member_id) {
+                AccountInfoByTokenAndMember::<T>::mutate(token_id, creator_member_id, |account_data| {
+                    account_data.amount = account_data.amount.saturating_add(creator_amount)
+                });
+            }
+
+            let user_amount = amount - creator_amount;
```

Comment:
```suggestion
            let user_amount = amount.saturating_sub(creator_amount);
```
That's a shame that `deny(clippy::integer_arithmetic)` doesn't catch that :/
We will probably need to use https://github.com/rust-lang/rust-clippy/pull/9130 in the furute.